esphome/components/sensor/filter.cpp

```diff
@@ -332,6 +332,40 @@ optional<float> ThrottleFilter::new_value(float value) {
   return {};
 }
 
+// ThrottleWithPriorityFilter
+ThrottleWithPriorityFilter::ThrottleWithPriorityFilter(uint32_t min_time_between_inputs,
+                                                       std::vector<TemplatableValue<float>> prioritized_values)
+    : min_time_between_inputs_(min_time_between_inputs), prioritized_values_(std::move(prioritized_values)) {}
+
+optional<float> ThrottleWithPriorityFilter::new_value(float value) {
+  bool is_prioritized_value = false;
+  int8_t accuracy = this->parent_->get_accuracy_decimals();
+  float accuracy_mult = powf(10.0f, accuracy);
+  const uint32_t now = millis();
```

Comment:
Can this use the cached time?
```suggestion
  const uint32_t now = App.get_loop_component_start_time()
```

Reply:
[`5081da5` (#9937)](https://github.com/esphome/esphome/pull/9937/commits/5081da5f19a960d7285e6b7e54e3528e577f9ce6)